### PR TITLE
Feat/display mode

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

SearchBar must have an option to control the display mode of the search and clear icons.

https://app.clubhouse.io/vtex/story/31177/search-bar-must-have-option-to-keep-the-search-icon

#### How should this be manually tested?

All three display modes are here:
https://kiwisearch--storecomponents.myvtex.com/

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/__tests__/components/SearchBar.test.js [hunk 1]
```diff
@@ -35,10 +38,47 @@ describe('<SearchBar />', () => {
   })
 
   it('should have CSS handle searchBarIcon', () => {
-    const { container } = renderComponent()
-
+    const { container, debug } = renderComponent()
```
Comment (steckhelena) [inline react/__tests__/components/SearchBar.test.js]: All these `debug` are not being used? If so I don't think they should be declared.

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 2]
```diff
@@ -64,35 +89,80 @@ const AutocompleteInput = ({
...
-        <CloseIcon />
-      ) : (
-        !hasIconLeft && <SearchIcon />
-      )}
+      <SearchIcon />
     </button>
   )
 
+  const externalSearchButton = showExternalSearchButton && (
+    <div className={`bw1 bl b--muted-4 flex items-center `}>
```
Comment (victorhmp) [inline react/components/SearchBar/components/AutocompleteInput.js]: Maybe add a CSS handle here

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 3]
```diff
@@ -64,35 +89,80 @@ const AutocompleteInput = ({
...
+        )}  flex items-center h-100 pointer pv0 nr5 ph5 bn c-link`}
+        onClick={onGoToSearchPage}
+      >
+        <SearchIcon />
+      </button>
+    </div>
+  )
+
+  const suffix = (
+    <div className="flex h-100">
```
Comment (victorhmp) [inline react/components/SearchBar/components/AutocompleteInput.js]: Also might be a good idea to add a CSS handle

File: react/__tests__/components/SearchBar.test.js (hunk 1 above)
Comment (kaisermann) [reply react/__tests__/components/SearchBar.test.js]: removed them 🙏

**Pull request merged**